astropy/stats/lombscargle/implementations/chi2_impl.py

```diff
@@ -0,0 +1,88 @@
+from __future__ import print_function, division
+
+import numpy as np
+
+from .mle import design_matrix
+
+
+def lombscargle_chi2(t, y, dy, frequency, normalization='standard',
+                     fit_bias=True, center_data=True, nterms=1):
+    """Lomb-Scargle Periodogram
+
+    This implements a chi-squared-based periodogram, which is relatively slow
+    but useful for validating the faster algorithms in the package.
+
+    Parameters
+    ----------
+    t, y, dy : array_like (NOT astropy.Quantities)
+        times, values, and errors of the data points. These should be
+        broadcastable to the same shape.
+    frequency : array_like
+        frequencies (not angular frequencies) at which to calculate periodogram
+    normalization : string (optional, default='standard')
+        Normalization to use for the periodogram.
+        Options are 'standard', 'model', 'log', or 'psd'.
+    fit_bias : bool (optional, default=True)
+        if True, include a constant offet as part of the model at each
+        frequency. This can lead to more accurate results, especially in the
+        case of incomplete phase coverage.
+    center_data : bool (optional, default=True)
+        if True, pre-center the data by subtracting the weighted mean
+        of the input data. This is especially important if ``fit_bias = False``
+    nterms : int (optional, default=1)
+        Number of Fourier terms in the fit
+
+    Returns
+    -------
+    power : array_like
+        Lomb-Scargle power associated with each frequency.
+        Units of the result depend on the normalization.
+
+    References
+    ----------
+    .. [1] M. Zechmeister and M. Kurster, A&A 496, 577-584 (2009)
+    .. [2] W. Press et al, Numerical Recipies in C (2002)
+    .. [3] Scargle, J.D. 1982, ApJ 263:835-853
+    """
+    if dy is None:
+        dy = 1
```

Comment:
In principle, since we know `t`, `y` are validated arrays here, this could just be
```
dy = broadcast_to(1., y.shape)
```
(need to import this above)
With that, the next line can go.  But really, also fine to keep as is; overhead will be small and this way the routine can be used on its own more easily.